kahuna/public/js/search/results.js

```diff
@@ -86,8 +90,23 @@ results.controller('SearchResultsCtrl', [
             lastSearchFirstResultTime = undefined;
         }
 
+        function initialSearch() {
+            /*
+            Maintain `lastSearchFirstResultTime` regardless of sorting order.
+
+            If we're sorting in ascending order, we need to get the upload time of the
+            last image, so we make a request for 1 image, then make a further request
+            where the offset is the total images - 1 from the initial request.
+             */
+            return angular.isUndefined(ctrl.filter.orderBy) ?
+                ctrl.searched = search({length: 1}) :
+                ctrl.searched = search({length: 1}).then((images) => {
```

Comment:
As mentioned orally, small risk of race condition here if the new results have come in before the second search, although we don't think it's break the UI.